src/httprpc.cpp

```diff
@@ -199,10 +199,8 @@ static bool HTTPReq_JSONRPC(const std::any& context, HTTPRequest* req)
                 req->WriteReply(HTTP_FORBIDDEN);
                 return false;
             }
-            UniValue result = tableRPC.execute(jreq);
 
-            // Send reply
-            strReply = JSONRPCReply(result, NullUniValue, jreq.id);
+            reply = JSONRPCExecOne(jreq, valRequest);
```

Comment:
In commit "rpc: refactor single/batch requests" (125ef7a61885d2ffc0d32b66cdfb3dced2a4417b)

This change doesn't seem right because `jreq.parse` was just called above on line 196, and now the request is going to be parsed for a second time inside `JSONRPCExecOne` for no reason. I think this could be fixed by just dropping the `jreq.parse` call inside `JSONRPCExecOne`. It looks like later commit 4f22286c4aa65a113c44153847c857075a4225d6 does this anyway, so would be better to drop it up front and not change semantics multiple times. It would also be good to make the `ExecOne` `jreq` parameter const at that point since it should no longer be modifying the request. I would also rename `ExecOne` and `ExecBatch` functions to avoid giving the misleading impression that they are alternatives to each other, when actually they are operating at different levels and do not do the same things. ExecBatch handles errors, ExecOne does not. ExecBatch parses and executes, ExecOne should only execute requests without parsing them. Actually, I would probably just drop the ExecBatch function entirely since it is short and only called in one place. It would be clearer to just inline, IMO.

Reply:
Great review, thank you! I cleaned up the commit history in a rebase. The [diff is minimal](https://github.com/bitcoin/bitcoin/compare/576ec3e2debdb63e142eee2c2fa00af995542f42..5d5e8cbca785db851747c5143dc1d651a1ea11e6) but in the range diff you can see the rebasing removed a commit: 

`git range-diff fbcf1029a7ba47d07a4566cf1f9d2e7b4870304a 576ec3e2debdb63e142eee2c2fa00af995542f42 5d5e8cbca785db851747c5143dc1d651a1ea11e6`

Regarding naming and in-lining, I could see the two functions being named `JSONRPCParseBatch()` and `JSONRPCExec()`? I still like both of those definitions in server.cpp but if you think the "parse batch" logic can live happily ever after in `HTTPReq_JSONRPC()` I won't protest. Either way would these 1 or 2 changes make more sense as a nice easy commit on top? Or should be rebased into the branch.